File: backend/hiFes/src/main/java/hiFes/hiFes/dto/postDto/PostDto.java
package hiFes.hiFes.dto.postDto;


import hiFes.hiFes.domain.BaseEntity;
import hiFes.hiFes.domain.Post;
import hiFes.hiFes.dto.commentDto.CommentDto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;
import java.util.stream.Collectors;

@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class PostDto {
    private Long id;
    private Long createdBy;
    private String writer;

    private String title;
    private String content;
    private String postType;
    private Boolean isHidden;
    private String hideReason;

    private String imagePath;

    private Long organizedFestivalId;

    private int views;
    private Float rating;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    private List<CommentDto> topLevelComments;
    private int commentsCount;


}

File: frontend/android/app/src/main/java/com/ssafy/hifes/data/model/PostDto.kt
package com.ssafy.hifes.data.model

data class PostDto(
    var id: Int,
    var createdBy: Int,
    var writer: String,
    var title: String,
    var postType: String,
    var organizedFestivalId: Int,
    var views: Int,
    var createdAt: DateTime,
    var content: String="",
    var isHidden: Boolean?,
    var hideReason: String?,
    var imagePath: String?,
    var rating: Float?,
    var commentsCount: Int = 0
)

data class PostDetailDto(
    var id: Int,
    var createdBy: Int,
    var writer: String,
    var title: String,
    var content: String,
    var postType: String,
    var isHidden: Boolean?,
    var imagePath: String?,
    var organizedFestivalId: Int,
    var views: Int,
    var rating: Float?,
    var createdAt: DateTime,
    var topLevelComments: List<CommentDto>
)
